tools/lint/lint.py

```diff
@@ -286,25 +287,18 @@ def check_owners(repo_root, paths):
             path = path.replace("\\", "/")
 
         source_file = SourceFile(repo_root, path, "/")
-        if source_file.name_is_non_test:
-            continue
-
         parts = source_file.dir_path.split(os.path.sep)
-        if (parts[0] in source_file.root_dir_non_test or
-            any(item in source_file.dir_non_test for item in parts) or
-            any(parts[:len(non_test_path)] == list(non_test_path) for non_test_path in source_file.dir_path_non_test)):
-            continue
         level = 0
         for dir in owner_in_subdirs:
-            if path.find(dir) == 0:
-                level = len(dir.split("/"))
-        if len(parts) <= level or len(parts) > level + 1:
+            if path.startswith(dir):
+                level = dir.count("/") + 1
+        if len(parts) <= level:
             continue
-        if not (parts[level]) in dirs:
-            dirs[parts[level]] = False
-        offset = path.find("/OWNERS")
-        if offset != -1:
-            dirs[parts[level]] = True
+        key = "/".join(parts[:level + 1])
+        if not key in dirs:
```

Comment:
We should be able to move the following if statement up here, and then have this as an `elif` branch.